components/google-cloud/google_cloud_pipeline_components/experimental/hyperparameter_tuning_job/component.py

```diff
@@ -0,0 +1,264 @@
+import json
+
+from kfp.components import create_component_from_func_v2
+from typing import NamedTuple
+
+def hyperparameter_tuning_job_run_op(
+    display_name: str,
+    project: str,
+    staging_bucket: str,
+    worker_pool_specs: list,
+    metric_spec: dict,
+    parameter_spec: dict,
+    max_trial_count: int,
+    parallel_trial_count: int,
+    max_failed_trial_count: int = 0,
+    location: str = "us-central1",
+    search_algorithm: str = None,
+    labels: dict = None,
+    measurement_selection: str = "best",
+    encryption_spec_key_name: str = None,
+    base_output_dir: str = None,
+) -> NamedTuple('Outputs', [
+    ("trials", list),
+]):
+    """
+    Creates a Google Cloud AI Plaform HyperparameterTuning Job.
+
+    Example usage:
+    ```
+    from google.cloud.aiplatform import hyperparameter_tuning as hpt
+    from google_cloud_pipeline_components.experimental import hyperparameter_tuning_job
+    from kfp.v2 import dsl
+
+    worker_pool_specs = [
+            {
+                "machine_spec": {
+                    "machine_type": "n1-standard-4",
+                    "accelerator_type": "NVIDIA_TESLA_K80",
+                    "accelerator_count": 1,
+                },
+                "replica_count": 1,
+                "container_spec": {
+                    "image_uri": container_image_uri,
+                    "command": [],
+                    "args": [],
+                },
+            }
+        ]
+    parameter_spec = hyperparameter_tuning_job.serialize_parameter_spec({
+        'lr': hpt.DoubleParameterSpec(min=0.001, max=0.1, scale='log'),
+        'units': hpt.IntegerParameterSpec(min=4, max=128, scale='linear'),
+        'activation': hpt.CategoricalParameterSpec(values=['relu', 'selu']),
+        'batch_size': hpt.DiscreteParameterSpec(values=[128, 256], scale='linear')
+    })
+
+
+    @dsl.pipeline(pipeline_root='', name='sample-pipeline')
+    def pipeline():
+        hp_tuning_task = hyperparameter_tuning_job.HyperparameterTuningJobRunOp(
+            display_name='hp-job',
+            project='my-project',
+            location='us-central1',
+            worker_pool_specs=worker_pool_specs,
+            metric_spec={
+                'loss': 'minimize',
+            },
+            parameter_spec=parameter_spec,
+            max_trial_count=128,
+            parallel_trial_count=8,
+            labels={'my_key': 'my_value'},
+        )
+
+    ```
+
+    For more information on using hyperparameter tuning please visit:
+    https://cloud.google.com/ai-platform-unified/docs/training/using-hyperparameter-tuning
+
+    Args:
+        display_name (str):
+            Required. The user-defined name of the HyperparameterTuningJob.
+            The name can be up to 128 characters long and can be consist
+            of any UTF-8 characters.
+        project (str):
+            Required. Project to run the HyperparameterTuningJob in.
+        staging_bucket (str):
```

Comment:
Is this CustomJobSpec.base_output_directory?
Is there any public HP tuning doc call this staging_bucket?

Reply:
Yes, this is basically CustomJobSpec.base_output_directory. It's required (either here or in aiplatform.init) for constructing a CustomJob. In aiplatform.init, it's called staging_bucket.

Constructing a CustomJob is required for constructing a HyperparameterTuningJob, but since staging_bucket is not directly required in HyperparameterTuningJob, there's no public HP tuning doc that calls this staging_bucket, but it's called staging_bucket in docs for [CustomJob](https://github.com/googleapis/python-aiplatform/blob/master/google/cloud/aiplatform/jobs.py#L949) and [aiplatform.init](https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/initializer.py#L60).

Reply:
Done. I've removed staging_bucket, using base_output_directory for both aiplatform.init and  in CustomJob.